lib/datastore/request.js

```diff
@@ -469,11 +471,21 @@ DatastoreRequest.prototype.runQuery = function(query, callback) {
     if (notFinished) {
       // Run the query again to make sure all of the requested entities are
       // returned.
+      if (originalLimitVal > -1) {
+        // Update the limit on the nextQuery to return only the amount of
+        // results originally asked for.
+        nextQuery.limit(originalLimitVal - entities.length);
```

Comment:
It may make sense to track the `originalLimitVal` only for your `nextQuery` computation and use this instead (not sure about getting the size of a list from the proto in node)

    nextQuery.limit(query.limit - resp.batch.entityResultSize)